### PR TITLE
Add support for incremental loads to FDW plugins

### PR DESCRIPTION
**Pull request by mildbyte**
Add a `cursor_columns` field to the table parameters (used as a list
of columns that form an increasing-only replication bookmark). Store the
ingestion state in a table inside of the image, similarly to Airbyte.

### CODE REVIEW

File: splitgraph/hooks/data_source/fdw.py [hunk 1]
```diff
@@ -299,6 +344,119 @@ def _load(self, schema: str, tables: Optional[TableInfo] = None):
...
+
+                repository.objects.record_table_as_base(
+                    repository,
+                    table_name,
+                    new_image_hash,
+                    chunk_size=DEFAULT_CHUNK_SIZE,
+                    source_schema=staging_schema,
+                    source_table=table_name,
+                    table_schema=new_schema,
+                )
```
Comment (mildbyte) [inline splitgraph/hooks/data_source/fdw.py:428]: There's some overlap between this code and the code in https://github.com/splitgraph/splitgraph/blob/master/splitgraph/ingestion/airbyte/data_source.py#L290-L329 but I don't know if there's a nice way to factor it out + when writeable LQ lands, we should be able to simplify this code anyway by writing into the LQ checkout.

Comment (gruuya) [reply splitgraph/hooks/data_source/fdw.py:428]: Yeah, makes sense.

**Review by gruuya — APPROVED** [review]
Looks good!

File: splitgraph/engine/base.py [hunk 2]
```diff
@@ -134,22 +135,42 @@ def copy_table(
         target_schema: str,
         target_table: str,
         with_pk_constraints: bool = True,
+        cursor_fields: Optional[Dict[str, str]] = None,
```
Comment (Gelio) [inline splitgraph/engine/base.py:138]: image


image

image

Array fields are not supported by dynamic forms. I suppose I should relax the error message

**Pull request merged**